core/src/main/java/tech/tablesaw/io/ReadOptions.java

```diff
@@ -75,6 +75,7 @@
   protected final String missingValueIndicator;
   protected final boolean minimizeColumnSizes;
   protected final int maxCharsPerColumn;
+  protected final int sampleRowsIfGreaterThan;
```

Comment:
I don't think it's appropriate to put this in `ReadOptions` because it doesn't apply to all readers. It would be confusing if you tried to set this on an `HtmlReader` or `JsonReader` for example and then it did nothing. I would put it just on `CsvReadOptions` for now

Reply:
Agreed. I somehow thought this would work for other readers. I'll fix this.

Reply:
The problem here is that `FileReader` only has access to `ReadOptions`, not `CsvReadOptions`. So I need to either implement sampling for all readers or introduce a new intermediary class `FileReaderOptions` in the options class hierarchy that is extended by `CsvReadOptions` and `FixedWidthReadOptions`.

Reply:
Perhaps you could just add a parameter to the `FileReader` method instead? I think introducing a `FileReadOptions` might end up being a lot of code and also be confusing (e.g. you could read json or html from a file, but they wouldn't use `FileReadOptions`

Reply:
I followed your suggestion and moved sampling to CsvReadOptions using a method parameter.